### ISSUE TITLE
[a11y]: Side panel with steps does not announce changes when content updates

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#7905]
### Package

@carbon/ibm-products

### Package version

v2.71.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

This is specific to [side panels with multiple steps](https://ibm-products.carbondesignsystem.com/?path=/story/components-sidepanel--panel-with-second-step).

Content modification / focus jump after activating View All button in the first step.

Users are unclear what occurred after activating the View All button.
The side panel content is updated in the panel and focus moves back to the top of the panel, but this unexpected for screen reader users.

**Recommended fix**
Avoid unnecessary surprises and make the web more predictable for assistive aid users:
1. Pre-inform the element behavior (content modified above) to screen reader users, by using more descriptive Labels, or users hints
2. Once the View all button is activated, move focus to the newly appeared  Back Button on the top of the panel.

**User impact**
Any unexpected behavior can confuse and disorient screen reader users who have low vision or cognitive disabilities.

_May also need to look into how we can announce more of the content changes._

### WCAG 2.1 Violation

2.4.3 Focus Order

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/components-sidepanel--panel-with-second-step

### Steps to reproduce

To reproduce this defect:
1. Use a screen reader and navigate in the side panel with second step story.
2. Navigate down to the View All button and activate it.

Upon activating the view all, the content changes, and the focus moves to Close.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
